### PR TITLE
add new keymap for pteron

### PR DESCRIPTION
**Pull request by alzafacon**
## Description

for elite-c controller
with diode direction override

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/pteron/keymaps/FSund/config.h [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/pteron/keymaps/FSund/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/pteron/keymaps/alzafacon/config.h [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/pteron/keymaps/alzafacon/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/pteron/keymaps/alzafacon/keymap.c [hunk 3]
```diff
@@ -0,0 +1,104 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/handwired/pteron/keymaps/alzafacon/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/handwired/pteron/keymaps/alzafacon/rules.mk [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no
+UNICODE_ENABLE = no
+UNICODEMAP_ENABLE = no
```
Comment (drashna) [inline keyboards/handwired/pteron/keymaps/alzafacon/rules.mk:27]: Most of this isn't needed.  Just what has been changed.

File: keyboards/handwired/pteron/keymaps/default/config.h [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
```
Comment (drashna) [inline keyboards/handwired/pteron/keymaps/default/config.h:4]: This should go in the keyboard config, not here.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/pteron/keymaps/default/config.h (hunk 5 above)
Comment (alzafacon) [reply keyboards/handwired/pteron/keymaps/default/config.h:4]: Besides adding a new keymap I am also making the diode direction configurable for each keymap. that is why I am adding DIODE_DIRECTION to each of the keymaps.

Is the default keymap special in some way that I shouldn't add the DIODE_DIRECTION override here?

Comment (drashna) [reply keyboards/handwired/pteron/keymaps/default/config.h:4]: It's less of the keymap, and more of ... if it's compiled on the configurator, it won't have this set, at all. 

If you need this set, then undefining it, and then defining the correct value in the keymap folder would be the correct way to handle this.

**Review by drashna — COMMENTED** [review]
Also, could you rename the files to be lower case.

File: keyboards/handwired/pteron/keymaps/default/config.h (hunk 5 above)
Comment (alzafacon) [reply keyboards/handwired/pteron/keymaps/default/config.h]: got it, done!

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/pteron/keymaps/alzafacon/keymap.c [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+ * +-------------+------+------+------+------|      |------+------+------+------+-------------+
+ *               |Lower | SPC  | Alt  | GUI  |      | Alt  | GUI  | SPC  |Raise |
+ *               +---------------------------+      +---------------------------+
+ */
+[_QWERTY] = LAYOUT( \
+  KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,            KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,            KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,            KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,  \
+                    LOWER,   KC_SPC,  KC_LALT, KC_LGUI,         KC_RALT, KC_RGUI, KC_SPC,  RAISE \
```
Comment (fauxpark) [inline keyboards/handwired/pteron/keymaps/alzafacon/keymap.c:50]:
```suggestion
[_QWERTY] = LAYOUT(
  KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,            KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,            KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,            KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
                    LOWER,   KC_SPC,  KC_LALT, KC_LGUI,         KC_RALT, KC_RGUI, KC_SPC,  RAISE
```

File: keyboards/handwired/pteron/keymaps/alzafacon/keymap.c [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+ * +-------------+------+------+------+------|      |------+------+------+------+-------------+
+ *               |      |      |      |      |      | Next | Vol- | Vol+ | Play |
+ *               +---------------------------+      +---------------------------+
+ */
+[_LOWER] = LAYOUT( \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,         KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,         KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,           KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,          KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+                    _______, _______, _______, _______,         KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
Comment (fauxpark) [inline keyboards/handwired/pteron/keymaps/alzafacon/keymap.c:71]:
```suggestion
[_LOWER] = LAYOUT(
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,         KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC,         KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,           KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,          KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
                    _______, _______, _______, _______,         KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

File: keyboards/handwired/pteron/keymaps/alzafacon/keymap.c [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+ * +-------------+------+------+------+------|      |------+------+------+------+-------------+
+ *               |      |      |      |      |      | Next | Vol- | Vol+ | Play |
+ *               +---------------------------+      +---------------------------+
+ */
+[_RAISE] = LAYOUT( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,           KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,          KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+                    _______, _______, _______, _______,         KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
Comment (fauxpark) [inline keyboards/handwired/pteron/keymaps/alzafacon/keymap.c:92]:
```suggestion
[_RAISE] = LAYOUT(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,            KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,           KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,          KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
                    _______, _______, _______, _______,         KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

File: keyboards/handwired/pteron/keymaps/alzafacon/keymap.c [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+ * +-------------+------+------+------+------|      |------+------+------+------+-------------+
+ *               |      |      |      |      |      |      |      |      |      |      
+ *               +---------------------------+      +---------------------------+
+ */
+[_ADJUST] = LAYOUT( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,           KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  _______, RESET,   DEBUG,   _______, _______, _______,         _______, _______, _______, _______, _______, KC_DEL,  \
+  _______, _______, _______, _______, _______, _______,         _______, _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______,         _______, _______, _______, _______, _______, _______, \
+                    _______, _______, _______, _______,         _______, _______, _______, _______  \
```
Comment (fauxpark) [inline keyboards/handwired/pteron/keymaps/alzafacon/keymap.c:113]:
```suggestion
[_ADJUST] = LAYOUT(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,           KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  _______, RESET,   DEBUG,   _______, _______, _______,         _______, _______, _______, _______, _______, KC_DEL,
  _______, _______, _______, _______, _______, _______,         _______, _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______,         _______, _______, _______, _______, _______, _______,
                    _______, _______, _______, _______,         _______, _______, _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by alzafacon** [comment]
@fauxpark  can you take a look again, please?

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**